### DISCUSSION TITLE
Parsing a yaml file

### DISCUSSION THREAD

**Discussion by ghost** [tighten/jigsaw#633] [General]
I've been trying to parse a file with jigaw, but doesn't seem to work.  
```
<?php
use Symfony\Component\Yaml\Yaml;

function platforms() {
	$data = Yaml::parseFile( '/srv/.global/custom.yml' );
}
```
the file i'm trying to grab is not in the same project folder but is at a different location.... any feedback would be great.

**Comment by bakerkretzmar** [comment]
What is the issue exactly? Are you getting an error message? Can you verify that the file exists (in the code, with a `file_exists()` right before you try to parse it or something)?

**Reply by ghost** [reply]
so the issue is, that the file exists over at /srv/.global/custom.yml, the way i did it shoudl work but it throws an error that the file is not found.

**Reply by ghost** [reply]
In Parser.php line 55:
                                                                               
  File "/srv/.global/custom.yml" does not exist. (View: /Users/benlumia007/.d  
  ev/sites/dashboard/domains/dev/public_html/public/_views/home.blade.php)     
                                                                               

In Parser.php line 55:
                                                  
  File "/srv/.global/custom.yml" does not exist.  
                                                  

build [--pretty PRETTY] [-c|--cache [CACHE]] [--] [<env>]

**Reply by bakerkretzmar** [reply]
I'm on a Mac and there is no root `/srv` directory, could you be looking for a file that exists on the server where your site is deployed but not on your local machine? What does `file_exists()` tell you?

**Reply by ghost** [reply]
so just to point out that ths /srv/ is from the docker container that hat is mounted as a volume. so i know it exists.

**Reply by ghost** [reply]
my other CMS platforms all work except for Jigsaw.

**Reply by bakerkretzmar** [reply]
Okay but it might not exist when/where Jigsaw is running. That error message indicates that it's looking for a path at `/srv` but the actual exception is coming from a Blade view that appears to be on your machine under `/Users`.

Can you try checking with `file_exists()`?

**Reply by ghost** [reply]
how would i use file_exiss in this case.

**Reply by ghost** [reply]
i see what you're saying, it grabs the hosts location... is there. away for to grab from the the container.. because what if i'm on a linux or windows patform.

**Reply by ghost** [reply]
This definitely works
```
<?php
function platforms() {
	$data = Symfony\Component\Yaml\Yaml::parseFile( '/Users/benlumia007/.dev/.global/custom.yml' );

	return $data
}
```

**Reply by bakerkretzmar** [reply]
For now you should make sure you're running Jigsaw in the same place that file lives. If the file is in the Docker container, run Jigsaw inside the Docker container, etc.

In general I would put that file inside your Jigsaw project and use a relative path, then it'll always be exactly where you expect.

**Reply by ghost** [reply]
make sense.. i'll do some experiments. Thanks

**Reply by ghost** [reply]
would this be an issue too cuz technically it should be able to parse the file inside and outside the container if the volumes are set. it just sounds like a bug

**Reply by bakerkretzmar** [reply]
Not sure sorry, I'm not very familiar with Docker.

**Reply by ghost** [reply]
well thanks anyways, I think i found a solution.
